From 8b4c1bd074b5214a508c2830575a70e6bd73132e Mon Sep 17 00:00:00 2001
From: Wentao Guan <guanwentao@uniontech.com>
Date: Tue, 7 Jan 2025 11:42:36 +0800
Subject: [PATCH 309/360] deepin: config: Enable NO_HZ_FULL

There is reportedly no (longer) significant overhead to this when it is
not enabled at runtime, see
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=804857#66

By default, without passing the nohz_full parameter, this behaves just
like NO_HZ_IDLE(default), enable it by passing nohz_full=all or nohz_full=cpulist, see
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/kernel/time/Kconfig?h=v6.6#n142
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/admin-guide/kernel-parameters.rst?h=v6.6#n80

Signed-off-by: Wentao Guan <guanwentao@uniontech.com>
---
 arch/arm64/configs/deepin_arm64_desktop_defconfig         | 1 +
 arch/loongarch/configs/deepin_loongarch_desktop_defconfig | 1 +
 arch/x86/configs/deepin_x86_desktop_defconfig             | 1 +
 3 files changed, 3 insertions(+)

diff --git a/arch/arm64/configs/deepin_arm64_desktop_defconfig b/arch/arm64/configs/deepin_arm64_desktop_defconfig
index 25a3f847aa3e..e3781ad54c49 100644
--- a/arch/arm64/configs/deepin_arm64_desktop_defconfig
+++ b/arch/arm64/configs/deepin_arm64_desktop_defconfig
@@ -5,6 +5,7 @@ CONFIG_DEFAULT_HOSTNAME=""
 CONFIG_SYSVIPC=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_USELIB=y
+CONFIG_NO_HZ_FULL=y
 CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
 CONFIG_BPF_JIT=y
diff --git a/arch/loongarch/configs/deepin_loongarch_desktop_defconfig b/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
index 992dcc78d4cf..4bfbb1b8d832 100644
--- a/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
+++ b/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
@@ -6,6 +6,7 @@ CONFIG_SYSVIPC=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_WATCH_QUEUE=y
 CONFIG_USELIB=y
+CONFIG_NO_HZ_FULL=y
 CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
 CONFIG_BPF_JIT=y
diff --git a/arch/x86/configs/deepin_x86_desktop_defconfig b/arch/x86/configs/deepin_x86_desktop_defconfig
index 03154de7fb37..33bde2f97a77 100644
--- a/arch/x86/configs/deepin_x86_desktop_defconfig
+++ b/arch/x86/configs/deepin_x86_desktop_defconfig
@@ -4,6 +4,7 @@ CONFIG_KERNEL_ZSTD=y
 CONFIG_SYSVIPC=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_USELIB=y
+CONFIG_NO_HZ_FULL=y
 CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
 CONFIG_BPF_JIT=y
-- 
2.20.1

